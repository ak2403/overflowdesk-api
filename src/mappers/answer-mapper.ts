import { Answer } from "../types/mappers/answer";
import { Mapper } from "../types/mappers/mapper";

export class AnswerMapper implements Mapper<Answer> {
  private _id!: number;
  private _body!: string;
  private _commentCount!: number;
  private _createdDate!: number;
  private _downVoteCount!: number;
  private _isAccepted!: boolean;
  private _lastActivityDate!: number;
  private _link!: string;
  private _ownerId!: number;
  private _questionId!: number;
  private _score!: number;
  private _upVoteCount!: number;

  set id(id: number) {
    this._id = id;
  }

  set body(body: string) {
    this._body = body;
  }

  set commentCount(count: number) {
    this._commentCount = count;
  }

  set createdDate(date: number) {
    this._createdDate = date;
  }

  set downVoteCount(count: number) {
    this._downVoteCount = count;
  }

  set isAccepted(accepted: boolean) {
    this._isAccepted = accepted;
  }

  set lastActivityDate(date: number) {
    this._lastActivityDate = date;
  }

  set link(link: string) {
    this._link = link;
  }

  set ownerId(id: number) {
    this._ownerId = id;
  }

  set questionId(id: number) {
    this._questionId = id;
  }

  set score(score: number) {
    this._score = score;
  }

  set upVoteCount(count: number) {
    this._upVoteCount = count;
  }

  build(): Answer {
    return {
      id: this._id,
      body: this._body,
      commentCount: this._commentCount,
      createdDate: this._createdDate,
      downVoteCount: this._downVoteCount,
      isAccepted: this._isAccepted,
      lastActivityDate: this._lastActivityDate,
      link: this._link,
      ownerId: this._ownerId,
      questionId: this._questionId,
      score: this._score,
      upVoteCount: this._upVoteCount,
    };
  }
}
