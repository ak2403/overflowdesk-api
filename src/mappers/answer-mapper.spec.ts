import { AnswerMapper } from "./answer-mapper";

describe("AnswerMapper()", () => {
  const answerMapper = new AnswerMapper();

  it("construct answer with passed attribute values", () => {
    answerMapper.id = 123;
    answerMapper.body = "sample body";
    answerMapper.commentCount = 2;
    answerMapper.downVoteCount = 1;
    answerMapper.questionId = 456;
    answerMapper.score = 2;
    answerMapper.upVoteCount = 2;

    expect(answerMapper.build()).toMatchSnapshot();
  });
});
