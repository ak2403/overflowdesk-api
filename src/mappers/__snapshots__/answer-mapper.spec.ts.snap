// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerMapper() construct answer with passed attribute values 1`] = `
{
  "body": "sample body",
  "commentCount": 2,
  "createdDate": undefined,
  "downVoteCount": 1,
  "id": 123,
  "isAccepted": undefined,
  "lastActivityDate": undefined,
  "link": undefined,
  "ownerId": undefined,
  "questionId": 456,
  "score": 2,
  "upVoteCount": 2,
}
`;
